File: Crio/StackAndQueue/14_MergeKSortedLists/MergeKSortedLists.py
from Solution import *

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    k = int(input())
    lis = list()
    for _ in range(k):
        _ = int(input())
        values = list(map(int, input().split()))
        head = createList(values)
        lis.append(head)
    ans = mergeKSortedLists(lis)
    while ans != None:
        print(ans.val,end = " ")
        ans = ans.next


if __name__ == "__main__":
    main()

File: Crio/StackAndQueue/14_MergeKSortedLists/MergeKSortedLists.java
import java.io.*;
import java.util.*;
import crio.ds.List.ListNode;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

class MergeKSortedLists {
    public static void main(String[] args) throws IOException{
        Scanner scanner = new Scanner(System.in);
        int k = scanner.nextInt();
        ListNode lists[] = new ListNode[k];
        for(int i = 0 ; i < k ; i++) {
            int n = scanner.nextInt();
            List<Integer>numbers = new ArrayList<Integer>();
            for(int j = 0 ; j < n ; j++) {
                int value = scanner.nextInt();
                numbers.add(value);
            }
            lists[i] = ListNode.createList(numbers);
        }
        scanner.close();
        ListNode result = new Solution().mergeKSortedLists(lists);
        while(result != null) {
            System.out.print(result.val);
            System.out.print(" ");
            result = result.next;
        }
    }
}
